main/integration-tests/__test__/api-tests/projects/update-project.test.js

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Update project scenarios', () => {
+  let setup;
+  let adminSession;
+  let defaultProject;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+    defaultProject = await adminSession.resources.projects.mustFind(setup.gen.defaultProjectId());
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Updating a project', () => {
+    it('should fail when non-admin user is trying to update project', async () => {
+      const testProjectId = setup.gen.string({ prefix: `non-admin-update-proj-test` });
+      const newProj = await adminSession.resources.projects.create({
+        id: testProjectId,
+        indexId: defaultProject.indexId,
+      });
+
+      const researcherSession = await setup.createResearcherSession({ projectId: [testProjectId] });
+      const updateBody = { rev: newProj.rev, description: setup.gen.description(), id: testProjectId };
+
+      await expect(
+        researcherSession.resources.projects.project(testProjectId).update(updateBody),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should pass when admin is trying to update project', async () => {
+      const testProjectId = setup.gen.string({ prefix: `admin-update-proj-test` });
+      const newProj = await adminSession.resources.projects.create({
+        id: testProjectId,
+        indexId: defaultProject.indexId,
+      });
+
+      const description = setup.gen.description();
+      const adminSession2 = await setup.createAdminSession({ projectId: [testProjectId] });
+      const updateBody = { rev: newProj.rev, description, id: testProjectId, indexId: defaultProject.indexId };
+
+      const retVal = await adminSession2.resources.projects.project(testProjectId).update(updateBody);
+      expect(retVal.id).toStrictEqual(testProjectId);
+      expect(retVal.description).toStrictEqual(description);
```

Comment:
I think if you combine these into one await expect().resolve.toMatchObject({ id: testProjectId, description });
We get better error messages from Jest if the update method fails.